### ISSUE TITLE
Exporting to macOS from Windows results in errors

### ISSUE THREAD

**Issue by EmberSkyMedia** [godot-jolt/godot-jolt#896]
Jolt seems to give an issue when exporting a project from Windows 10 Godot 4.2.2 to MacOS

![image](https://github.com/godot-jolt/godot-jolt/assets/109048742/77d804c5-f59f-4470-96e4-0c690e46aa2f)

Result is the following error on Mac (M1 chip)

![image](https://github.com/godot-jolt/godot-jolt/assets/109048742/7f45243e-c7da-45fb-92c2-5d82e6145707)

I made a basic build (1 scene - 1 static3D, 1 rigidebody3D, camera3d and corresponding mesh/collisions) to check it wasn't my larger game and the error persisted. (I also disabled the git-repo if that was an issue as symlinks are trying to link to things outside the res:// folder from my quick google).

**Comment by mihe** [comment]
I must admit I didn't even realize it was possible to export to macOS from other platforms.

I can confirm that I'm seeing the same error when doing so. I am however also seeing other strange errors in the Output log about MachO and LipO being unable to open the framework within the `.app`.

I have a feeling this is going to be a bug on the Godot side of things. This extension differ from the norm in that I don't ship just `.dylib` files but rather `.framework` folders with dylibs in them.

Anyway, I'll dig into and see where the problem is. Thank you for reporting the issue.

**Comment by mihe** [comment]
The message you're seeing regarding relative symlinks is apparently something Godot just emits for any non-Unix platform (i.e. Windows) when it copies extensions into the `.app` bundle, presumably because Windows typically doesn't deal with symlinks as well as Unix platforms do. Either way, it's not really relevant for this extension since there are no symlinks in the framework bundles, so you can safely ignore the message.

The problem you're having with your app bundle being "damaged" seems to have more to do with the signing of the app bundle. How exactly are you signing your app bundle from Windows at the moment, @EmberSkyMedia? I'm assuming you're using the "Built-in (ad-hoc only)" method? Does this actually work, even if you don't use/reference Godot Jolt? I would assume that Gatekeeper would stop you from loading this on pretty much any machine.

You'll never be able to properly sign/notarize your app bundle from Windows either way, since the only other option besides the built-in codesigning is to use `rcodesign`, but that doesn't support signing app bundles with dynamic libraries in them, so your only real option is to do the signing/notarization from a macOS machine, which should work just fine as-is.

**Comment by mihe** [comment]
> Does this actually work, even if you don't use/reference Godot Jolt?

To answer my own question, no it does not apparently. If you disable code-signing altogether in the export settings then macOS will report your app bundle as "damaged". So your options when exporting from Windows seems to be:

1. Disable code-signing, in which case it's considered "damaged" and won't run at all.
2. Use Godot's built-in/ad-hoc code-signing, but you will have to add an exclusion for the app in Gatekeeper, or disable Gatekeeper altogether, which seems ill-advised.
3. Use `rcodesign`, which won't work since it doesn't support dynamic libraries.

Even if you were to go with option 2, I found a couple of issues with the built-in code-signing in the context of projects that rely on GDExtension, which I made a PR for here: godotengine/godot#93101

All-in-all, this is not an issue with this extension, so I'll go ahead and close this.
